biblib/views/document_view.py

```diff
@@ -456,3 +542,364 @@ def delete(self, library):
+        for access_type in update_allowed:
+            if cls.helper_access_allowed(service_uid=service_uid,
+                                         library_id=library_id,
+                                         access_type=access_type):
+                return True
+
+        return False
+
+    @classmethod
+    def delete_access(cls, service_uid, library_id):
+        """
+        Defines which type of user has delete permissions to a library.
+
+        :param service_uid: the user ID within this microservice
+        :param library_id: the unique ID of the library
+
+        :return: boolean, access (True), no access (False)
+        """
+        delete_allowed = cls.helper_access_allowed(service_uid=service_uid,
+                                                   library_id=library_id,
+                                                   access_type='owner')
+        return delete_allowed
+
+    @classmethod
+    def write_access(cls, service_uid, library_id):
+        """
+        Defines which type of user has write permissions to a library.
+
+        :param service_uid: the user ID within this microservice
+        :param library_id: the unique ID of the library
+
+        :return: boolean, access (True), no access (False)
+        """
+
+        read_allowed = ['write', 'admin', 'owner']
+        for access_type in read_allowed:
+            if cls.helper_access_allowed(service_uid=service_uid,
+                                         library_id=library_id,
+                                         access_type=access_type):
+                return True
+
+        return False
+
+    @staticmethod
+    def library_name_exists(service_uid, library_name):
+        """
+        Checks to see if a library name already exists in the user's created
+        libraries
+
+        :param service_uid: the user ID within this microservice
+        :param library_name: name to check if it exists
+
+        :return: True (exists), False (does not exist)
+        """
+
+        with current_app.session_scope() as session:
+            library_names = \
+                [i.library.name for i in
+                 session.query(Permissions)\
+                     .filter_by(user_id = service_uid)\
+                     .filter(Permissions.permissions['owner'].astext.cast(Boolean).is_(True)).all()]
+
+        if library_name in library_names:
+            current_app.logger.error('Name supplied for the library already '
+                                     'exists: "{0}"'.format(library_name))
+
+            return True
+        else:
+            return False
+ 
+    def post(self, library):
+        """
+        HTTP POST request that adds a document to a library for a given user
+        :param library: library ID
+
+        :return: the response for if the library was successfully created
+
+        Header:
+        -------
+        Must contain the API forwarded user ID of the user accessing the end
+        point
+
+        Post body:
```

Comment:
This docstring still needs to be updated to reflect that the post body is a query, not a list of bibcodes

Reply:
cleaned up doctring